src/java/org/apache/cassandra/locator/Ec2MetadataServiceConnector.java

```diff
@@ -45,7 +45,7 @@ abstract class Ec2MetadataServiceConnector extends AbstractCloudMetadataServiceC
 
     Ec2MetadataServiceConnector(SnitchProperties properties)
     {
-        super(properties.get(EC2_METADATA_URL_PROPERTY, DEFAULT_EC2_METADATA_URL));
+        super(properties.get(EC2_METADATA_URL_PROPERTY, properties.get(METADATA_URL_PROPERTY, DEFAULT_EC2_METADATA_URL)));
```

Comment:
why do we need this?

Reply:
perhaps we should not bother that much - who would set it to something else than a default value?

Reply:
yeah ... true, hm.